### PR TITLE
fix(permissions): separate Video, Talks, and Tickets access permissions

### LINKED ISSUE #4044: Permission inconsistencies with Video, Talks, and Tickets access
## Description

There are a couple of significant inconsistencies regarding user access permissions that need to be addressed to ensure proper Role-Based Access Control (RBAC):

1. **Video and Talks Permission Overlap:** Currently, if a user or role is granted *only* "Video" permissions, they inadvertently gain full access to "Talks" as well. The video permission should be isolated and not implicitly grant access to the talks module unless explicitly assigned.
2. **Tickets Permission Instability:** The permission system governing the "Tickets" module is currently experiencing unexpected behavior and is not functioning as intended. This requires a thorough review and stabilization to ensure users are granted the correct access levels based on their roles.

## Expected Behavior

- Permissions should be strictly isolated. Granting access to "Video" should not have any side effects on "Talks".
- The "Tickets" module permissions should be stable, predictable, and accurately reflect the assigned roles.

## Acceptance Criteria

- [ ] Audit the permission check logic (both frontend and backend) for the Video module.
- [ ] Ensure that "Talks" permissions are verified independently from "Video" permissions.
- [ ] Investigate the root cause of the current instability in the Tickets permission system.
- [ ] Refactor or fix the Tickets permission logic to restore proper functionality.
- [ ] Add or update unit/integration tests to verify that these permission boundaries are strictly enforced.

### PR DESCRIPTION
**Pull request by Rachit7168**
## Description
Addresses inconsistencies in Role-Based Access Control (RBAC) across Video, Talks, and Tickets modules to ensure proper permission isolation.

### Changes Made:
1. **Gated Product, Category, and Quota views** under `control` with `EventPermissionRequiredMixin` and mapped them to the `can_change_items` permission to prevent direct URL access.
2. **Gated the control-side lazy widgets endpoint** (`event_index_widgets_lazy`) behind a `can_view_orders` check to prevent leaking order statistics.
3. **Filtered order-data widgets** (Attendee counts, Revenue, Waiting list, Check-ins) on the common-side event dashboard so they are hidden from users who lack the specific `can_view_orders` permission (even if they have generic ticket access like `can_change_items`).
4. **Restricted Pretalx (Talks) dashboard access** (`base.orga_access_event`) so it only allows users with talk-specific permissions (`can_change_submissions` or `is_reviewer`) or platform admins, blocking ticket-only users from accessing the talks backend.
5. **Added test cases** in `tests/tickets/control/test_permissions.py` covering product/quota views and lazy widget endpoints with and without permission.

Fixes #4044

https://github.com/user-attachments/assets/7d05c6cc-49c8-41ff-a203-458c1c479a40

## Summary by Sourcery

Align RBAC for tickets, dashboards, and talks so that order data and talk backends are only accessible with appropriate permissions.

Bug Fixes:
- Enforce can_change_items-based access control for product, category, question, and quota list/detail views to prevent unauthorized direct URL access.
- Restrict control-side lazy widgets endpoint to users with can_view_orders so order-related statistics cannot be leaked.
- Hide order-data widgets on the common event dashboard for users without can_view_orders, even if they have general ticket access.
- Constrain talks (Pretalx) dashboard access to users with talk-specific permissions or administrators, blocking ticket-only users from the talks backend.

Tests:
- Add permission tests for product/quota list views and lazy widget endpoint behavior with and without can_view_orders.

### CODE REVIEW

File: app/eventyay/talk_rules/event.py [hunk 1]
```diff
@@ -60,7 +60,11 @@ def can_change_organizer_settings(user, obj):
 def has_any_permission(user, obj):
     if not user or user.is_anonymous:
         return False
-    return bool(user.get_permissions_for_event(obj.event))
+    event = getattr(obj, 'event', None) or obj
+    if user.is_administrator:
+        return True
+    perms = user.get_permissions_for_event(event)
+    return 'can_change_submissions' in perms or 'is_reviewer' in perms
```
Comment (sarafarajnasardi) [inline app/eventyay/talk_rules/event.py:77]: This changes `has_any_permission()` from a generic “any event permission” predicate into a talk-only predicate. Since this backs `base.orga_access_event` and `base.view_event`, ticket-only users with permissions like `can_change_items`, `can_view_orders`, or `can_change_event_settings` will now fail generic event access checks even though they do have event permissions.
Could we keep `has_any_permission()` generic and use a separate talk-specific predicate/helper for hiding or allowing talk dashboard access?

**Review by sarafarajnasardi — CHANGES_REQUESTED** [review]

**Review by prayag78 — APPROVED** [review]

**Review by sarafarajnasardi — APPROVED** [review]
LGTM

**Comment by Rachit7168** [comment]
Video permission not working

**Comment by Rachit7168** [comment]
Will reopen once PR: #4750 is merged

**Review by Sak1012 — APPROVED** [review]

**Review by Saksham-Sirohi — APPROVED** [review]
